lark/lexer.py

```diff
@@ -158,20 +158,20 @@ class Token(str):
 
     def __new__(cls, type_, value, start_pos=None, line=None, column=None, end_line=None, end_column=None, end_pos=None):
         try:
-            self = super(Token, cls).__new__(cls, value)
+            inst = super(Token, cls).__new__(cls, value)
```

Comment:
I disagree with this naming change: After getting an instant via `super.__name__`, we are initializing it like a `__init__` would. No reason  not to treat it like that.

Reply:
Okay. I don't have a strong preference either way.

Reply:
Actually, let me try to push back by a little -

> we are initializing it like a __init__ would. 

Not quite, in `__init__` there is no need to return self.

It is also quite unusual to assign to self. In other language, where self/this are builtin, that would not be allowed.

Reply:
Yeah fair enough. I don't feel to strong either.